core/src/main/java/org/elasticsearch/search/rescore/QueryRescorerBuilder.java

```diff
@@ -56,7 +57,7 @@
     static {
         QUERY_RESCORE_PARSER.declareObject(InnerBuilder::setQueryBuilder, (p, c) -> {
             try {
-                return c.parseInnerQueryBuilder();
+                return c.parseInnerQueryBuilder().orElse(QueryBuilders.matchAllQuery());
```

Comment:
Allowing null here will later cause NPEs in QueryRescoreBuilder#build() during rewrite. We don't check this in QueryRescoreBuilders ctor yet, but I will add those checks. The current (2.x) behaviour for empty clauses as rescore query is to return all documents in the original query with score unchanged (provided the query/rescore weights are 1.0)